lib/cli-engine.js

```diff
@@ -670,15 +674,28 @@ class CLIEngine {
             // replace \ with / for Windows compatibility
             format = format.replace(/\\/g, "/");
 
+            let cwd = this.options ? this.options.cwd : process.cwd();
+
+            const namespace = naming.getNamespaceFromTerm(format);
+
+            if (namespace) {
+                cwd = `${cwd}/${namespace}`;
```

Comment:
I'm not sure this line is correct. Namespace shouldn't affect path (as far as I know), it's just a way to point to a specific package in NPM repository.

Reply:
It should change the working directory to "scope" of the scoped package. E.g. the package @foo/bar will be in node_modules/@foo/bar, so cwd should be node_modules/@foo so that bar can be resolved.

It's possible that I'm confused about something here, but when I install scoped packages this does indeed affect the path.

Reply:
Oh I see where I might have got confused. The module resolver should already be able to handle scoped packages, right? I only need to remove the namespace in order to enforce the prefix. The namespace should be added back when passing the format to the module resolver? That should simplify quite a bit I think.

Reply:
I have now updated the PR in an attempt to untangle this confusion. Hope that helps?